app/assets/javascripts/delay.js

```diff
@@ -0,0 +1,8 @@
+// http://stackoverflow.com/a/1909508/1314848
+var delay = (function(){
+  var timer = 0;
+  return function(callback, ms){
```

Comment:
I would rather use something like `delay_time` insead of `ms` and write some docs:

```js
// @param callback [Function] callback function, which is invoked after a given delay
// @param delay [Number] time in milliseconds until callback is invoked / triggered
return function(callback, ms){
```